python/cudf/groupby/groupby.py

```diff
@@ -259,9 +325,10 @@ def apply_multicolumn(self, result, aggs):
             # increase by 1 for every n values where n is the number of aggs
             # [['x,', 'z'], ['sum', 'min']]
             # codes == [[0, 1], [0, 1]]
-            code_size = max(len(aggs), len(self._val_columns))
-            codes.append(list(np.zeros(code_size, dtype='int64')))
-            codes.append(list(range(code_size)))
+            first_codes = [len(aggs) * [d*1] for d in range(len(
+                    self._val_columns))]
+            codes.append([item for sublist in first_codes for item in sublist])
+            codes.append(len(self._val_columns) * [0, 1])
             result.columns = MultiIndex(levels, codes)
```

Comment:
We should be able to take the `_source_data` path here, no? Or does that only exist for MultiIndex currently? This looks like another performance trap of manually doing dictionary encoding.

Reply:
This is actually just creating a MultiColumn version of the prefix method we used in legacy groupby. It creates a small set of levels and codes to represent the different levels of results when "max: a, max: b, min: a, min: b" etc is returned.